unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/States/FirstPerson/BuildFirstPersonMode.cs

```diff
@@ -0,0 +1,272 @@
+using DCL.Controllers;
+using DCL.Helpers;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BuildFirstPersonMode : BuildModeState
+{
+    [Header("Design variables")]
+    public float scaleSpeed = 0.25f;
+    public float rotationSpeed = 0.5f;
+    public float distanceFromCameraForNewEntitties = 5;
+
+
+    [Header("InputActions")]
+    [SerializeField] internal InputAction_Hold rotationHold;
+
+
+    Quaternion initialRotation;
+
+    float currentScaleAdded, currentYRotationAdded;
+
+    bool snapObjectAlreadyMoved = false,shouldRotate = false;
+    Transform originalParentGOEdit;
+
+    private void Start()
+    {
+        rotationHold.OnStarted += (o) => { shouldRotate = true; };
```

Comment:
Move this to delegates (as we did in another example) so we can unsubscribe properly